grails-app/services/org/pih/warehouse/data/ProductAssociationDataService.groovy

```diff
@@ -65,78 +65,38 @@ class ProductAssociationDataService {
-                command.errors.reject("Row ${index + 1}: Association already exists")
-            }
-
 
-            // find another product association that matches current associationInstance
-            // for two-way association relationship, where:
-            // - association.product = otherAssociation.associatedProduct
-            // - association.associatedProduct = otherAssociation.product
-            // - association.code = otherAssociation.code
-            def otherMatchingTwoWayAssociation = command.data.find {
-                it['product.productCode'] == params['associatedProduct.productCode'] &&
-                it['associatedProduct.productCode'] == params['product.productCode'] &&
-                it['code'] == params['code']
-            }
+            ProductAssociation existingAssociation = ProductAssociation.findWhere([
+                    product:            productAssociationInstance.product,
+                    associatedProduct:  productAssociationInstance.associatedProduct,
+                    code:               productAssociationInstance.code,
+            ])
 
-            if (otherMatchingTwoWayAssociation) {
-                // if such a product association exists then
-                // mark current association as two-way association
-                // and validate quantity on both product associations
-                params.hasMutualAssociation = true
-                if (otherMatchingTwoWayAssociation['quantity'] * productAssociationInstance.quantity != 1) {
-                    command.errors.reject(
-                            "Row ${index + 1}: Quantity of Product association for Products with codes " +
-                            "'${productAssociationInstance.product?.productCode}' and " +
-                            "'${otherMatchingTwoWayAssociation['associatedProduct.productCode']}' does not match"
-                    )
-                }
+            if (existingAssociation) {
+                command.errors.reject("Row ${index + 1}: association already exists")
             }
         }
     }
 
     void importData(ImportDataCommand command) {
         log.info "Import data " + command.filename
 
-        def individualAssociations = command.data.findAll{ !it.hasMutualAssociation }
-        // map of two-way product associations grouped by a product and associatedProduct pairs
-        def twoWayAssociations = command.data
-                .findAll{ it.hasMutualAssociation }
-                .groupBy { [it['product.productCode'], it['associatedProduct.productCode']] as Set<String> }
-        // create individual associations that are not bound by two-way association relationship
-        individualAssociations.each { params ->
+        command.data.each{ params ->
             ProductAssociation productAssociationInstance = new ProductAssociation(params)
             if (productAssociationInstance.validate()) {
+                if (params.hasMutualAssociation == "yes") {
```

Comment:
I think this should first look into `Boolean.valueOf(params.hasMutualAssociation)` and the potentially `|| params.hasMutualAssociation?.toLowerCase() == "yes")`